internal/loadbalancertype/data_source.go

```diff
@@ -0,0 +1,295 @@
+package loadbalancertype
+
+import (
+	"context"
+	_ "embed"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/datasourcevalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hetznercloud/hcloud-go/hcloud"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/util/datasourceutil"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/util/hcloudutil"
+)
+
+const (
+	// DataSourceType is the type name of the Hetzner Cloud load balancer type datasource.
+	DataSourceType = "hcloud_load_balancer_type"
+
+	// DataSourceListType is the type name of the Hetzner Cloud load balancer type list datasource.
+	DataSourceListType = "hcloud_load_balancer_types"
+)
+
+type resourceData struct {
+	ID                      types.Int64  `tfsdk:"id"`
+	Name                    types.String `tfsdk:"name"`
+	Description             types.String `tfsdk:"description"`
+	MaxAssignedCertificates types.Int64  `tfsdk:"max_assigned_certificates"`
+	MaxConnections          types.Int64  `tfsdk:"max_connections"`
+	MaxServices             types.Int64  `tfsdk:"max_services"`
+	MaxTargets              types.Int64  `tfsdk:"max_targets"`
+}
+
+var resourceDataAttrTypes = map[string]attr.Type{
+	"id":                        types.Int64Type,
+	"name":                      types.StringType,
+	"description":               types.StringType,
+	"max_assigned_certificates": types.Int64Type,
+	"max_connections":           types.Int64Type,
+	"max_services":              types.Int64Type,
+	"max_targets":               types.Int64Type,
+}
+
+func newResourceData(_ context.Context, in *hcloud.LoadBalancerType) (resourceData, diag.Diagnostics) { //nolint:unparam
+	var data resourceData
+	var diags diag.Diagnostics
+
+	data.ID = types.Int64Value(int64(in.ID))
+	data.Name = types.StringValue(in.Name)
+	data.Description = types.StringValue(in.Description)
+
+	data.MaxAssignedCertificates = types.Int64Value(int64(in.MaxAssignedCertificates))
+	data.MaxConnections = types.Int64Value(int64(in.MaxConnections))
+	data.MaxServices = types.Int64Value(int64(in.MaxServices))
+	data.MaxTargets = types.Int64Value(int64(in.MaxTargets))
+
+	return data, diags
+}
+
+func getCommonDataSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"id": schema.Int64Attribute{
+			Optional: true,
+			Computed: true,
+		},
+		"name": schema.StringAttribute{
+			Optional: true,
+			Computed: true,
+		},
+		"description": schema.StringAttribute{
+			Computed: true,
```

Comment:
Why is `description` the only non-optional attribute?

Reply:
All the fields except id and name should be read only, I updated this.